glean-core/rlb/src/private/labeled.rs

```diff
@@ -0,0 +1,366 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use inherent::inherent;
+use std::sync::Arc;
+
+use glean_core::metrics::MetricType;
+use glean_core::ErrorType;
+
+use crate::dispatcher;
+
+/// Sealed traits protect against downstream implementations.
+///
+/// We wrap it in a private module that is inaccessible outside of this module.
+mod private {
+    use crate::{
+        private::BooleanMetric, private::CounterMetric, private::StringMetric, CommonMetricData,
+    };
+    use glean_core::traits::ToLabeled;
+    use std::sync::Arc;
+
+    // `LabeledMetric<BooleanMetric>` is possible.
+    //
+    // See [Labeled Booleans](https://mozilla.github.io/glean/book/user/metrics/labeled_booleans.html).
+    impl ToLabeled for BooleanMetric {
+        type InnerCoreType = glean_core::metrics::BooleanMetric;
```

Comment:
Note to self: let's drop the `Type` suffix. This should either be `InnerCore` or `InnerCoreMetric`.